base/utf32.jl

```diff
@@ -20,15 +20,234 @@ length(s::UTF32String) = length(s.data) - 1
 function utf32(c::Integer...)
     a = Array(Char, length(c) + 1)
     for i = 1:length(c)
-        a[i] = Char(c[i])
+        @inbounds a[i] = Char(c[i])
     end
     a[end] = Char(0)
     UTF32String(a)
 end
 
+#=
+"""
+@brief      Converts a UTF-8 encoded string to UTF-32 encoding
+
+@param[in]  iStr::UTF8String
+
+@return     ::UTF32String
+@throws     UnicodeError
+
+            Hopefully copy! can be sped up to handle this case of widening
```

Comment:
I don't know if @JeffBezanson has improved copy! yet, it was a good deal slower when I first attempted this a few weeks ago.